homeassistant/components/stream/fmp4utils.py

```diff
@@ -2,67 +2,59 @@
 from __future__ import annotations
 
 from collections.abc import Generator
-import io
 
 
 def find_box(
-    segment: io.BytesIO, target_type: bytes, box_start: int = 0
+    mp4_bytes: bytes | memoryview, target_type: bytes, box_start: int = 0
 ) -> Generator[int, None, None]:
     """Find location of first box (or sub_box if box_start provided) of given type."""
     if box_start == 0:
-        box_end = segment.seek(0, io.SEEK_END)
-        segment.seek(0)
         index = 0
+        box_end = len(mp4_bytes)
     else:
-        segment.seek(box_start)
-        box_end = box_start + int.from_bytes(segment.read(4), byteorder="big")
+        box_end = box_start + int.from_bytes(
+            mp4_bytes[box_start : box_start + 4], byteorder="big"
+        )
         index = box_start + 8
     while 1:
         if index > box_end - 8:  # End of box, not found
             break
-        segment.seek(index)
-        box_header = segment.read(8)
+        box_header = mp4_bytes[index : index + 8]
         if box_header[4:8] == target_type:
             yield index
-            segment.seek(index)
         index += int.from_bytes(box_header[0:4], byteorder="big")
 
 
-def get_init(segment: io.BytesIO) -> bytes:
-    """Get init section from fragmented mp4."""
-    moof_location = next(find_box(segment, b"moof"))
-    segment.seek(0)
-    return segment.read(moof_location)
+def get_init_and_moof_data(segment: memoryview) -> tuple[bytes, bytes]:
```

Comment:
(Optional, for future change) I notice that get_init_and_moof_data is always passed to the constructor of a Segment. Maybe it would make sense to move this to be a "create_segment" method or something in a future PR.  Something to consider if you think it makes sense.

Reply:
No need to do this I think..this method will disappear in the next few PRs when we start splitting out the part data.